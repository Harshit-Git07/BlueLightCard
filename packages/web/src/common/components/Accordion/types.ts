export interface AccordionProps {
  title: string;
  children: React.ReactNode;
  onClickOpen?: () => void;
}
