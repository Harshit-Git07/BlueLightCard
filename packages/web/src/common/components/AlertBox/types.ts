export interface AlertBoxProps {
  alertType: string;
  title: string;
  description: any;
}

export interface style {
  textColor: string;
  backgroundColor: string;
}

export interface config {
  [key: string]: style;
}
