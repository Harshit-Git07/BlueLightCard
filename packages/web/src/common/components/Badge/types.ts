export type BgColorString = `bg-${string}`;

export type BadgeProps = {
  label: string;
  color: BgColorString;
  size: 'small' | 'large';
};
