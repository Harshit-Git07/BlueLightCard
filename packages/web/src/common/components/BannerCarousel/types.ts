import { CampaignCardProps } from '../CampaignCard/types';

export type BannerCarouselPropers = {
  banners: CampaignCardProps[];
};
