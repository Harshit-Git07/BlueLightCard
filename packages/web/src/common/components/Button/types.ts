import { ThemeVariant } from '@/types/theme';
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { MouseEventHandler, PropsWithChildren } from 'react';

export type ButtonProps = PropsWithChildren & {
  type?: 'button' | 'submit' | 'link';
  disabled?: boolean;
  slim?: boolean;
  className?: string;
  variant?: ThemeVariant;
  invertColor?: boolean;
  iconLeft?: IconDefinition;
  iconRight?: IconDefinition;
  href?: string;
  onClick?: MouseEventHandler;
};
