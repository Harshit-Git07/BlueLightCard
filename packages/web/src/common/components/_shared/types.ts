import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { ChangeEventHandler, ForwardedRef, MouseEventHandler, PropsWithChildren } from 'react';

export type InputFieldWrapperProps = PropsWithChildren & {
  icon?: IconDefinition;
  iconRight?: IconDefinition;
  showRightIcon?: boolean;
  showErrorState?: boolean;
  select?: boolean;
  showSuccessState?: boolean;
  onRightIconClick?: MouseEventHandler;
};

export interface InputFieldSharedProps<E, V = string> {
  name?: string;
  error?: boolean;
  success?: boolean;
  value?: V;
  required?: boolean;
  _ref?: ForwardedRef<unknown>;
  onChange?: ChangeEventHandler<E>;
}
