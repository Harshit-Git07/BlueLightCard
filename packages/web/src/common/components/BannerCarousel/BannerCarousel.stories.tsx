import { Meta, StoryFn } from '@storybook/react';
import BannerCarousel from './BannerCarousel';
import { CampaignCardProps } from '../CampaignCard/types';

const componentMeta: Meta = {
  title: 'Component System/BannerCarousel',
  component: BannerCarousel,
  argTypes: {
    bannerOne: {
      name: 'Show Banner One',
      control: 'boolean',
    },
    bannerOneName: {
      name: 'Banner One Name',
      control: 'text',
      if: {
        arg: 'bannerOne',
      },
    },
    bannerOneImage: {
      name: 'Banner One Image URL',
      control: 'text',
      if: {
        arg: 'bannerOne',
      },
    },
    bannerOneLink: {
      name: 'Banner One Link URL',
      control: 'text',
      if: {
        arg: 'bannerOne',
      },
    },
    bannerTwo: {
      name: 'Show Banner Two',
      control: 'boolean',
    },
    bannerTwoName: {
      name: 'Banner Two Name',
      control: 'text',
      if: {
        arg: 'bannerTwo',
      },
    },
    bannerTwoImage: {
      name: 'Banner Two Image URL',
      control: 'text',
      if: {
        arg: 'bannerTwo',
      },
    },
    bannerTwoLink: {
      name: 'Banner Two Link URL',
      control: 'text',
      if: {
        arg: 'bannerTwo',
      },
    },
    bannerThree: {
      name: 'Show Banner Three',
      control: 'boolean',
    },
    bannerThreeName: {
      name: 'Banner Three Name',
      control: 'text',
      if: {
        arg: 'bannerThree',
      },
    },
    bannerThreeImage: {
      name: 'Banner Three Image URL',
      control: 'text',
      if: {
        arg: 'bannerThree',
      },
    },
    bannerThreeLink: {
      name: 'Banner Three Link URL',
      control: 'text',
      if: {
        arg: 'bannerThree',
      },
    },
  },
  parameters: {
    docs: {
      description: {
        component:
          'This component combines the use of the SwiperCarousel and the CampaignCard to create a Banner Carousel for image offers.',
      },
    },
    controls: {
      exclude: ['banners'],
    },
  },
};

const DefaultTemplate: StoryFn = ({
  bannerOne,
  bannerOneName,
  bannerOneImage,
  bannerOneLink,
  bannerTwo,
  bannerTwoName,
  bannerTwoImage,
  bannerTwoLink,
  bannerThree,
  bannerThreeName,
  bannerThreeImage,
  bannerThreeLink,
}) => {
  const banners: CampaignCardProps[] = [];
  if (bannerOne) {
    banners.push({ image: bannerOneImage, linkUrl: bannerOneLink, name: bannerOneName });
  }
  if (bannerTwo) {
    banners.push({ image: bannerTwoImage, linkUrl: bannerTwoLink, name: bannerTwoName });
  }
  if (bannerThree) {
    banners.push({ image: bannerThreeImage, linkUrl: bannerThreeLink, name: bannerThreeName });
  }
  return <BannerCarousel banners={banners} />;
};

export const Default = DefaultTemplate.bind({});

Default.args = {
  bannerOne: true,
  bannerOneName: 'Banner One',
  bannerOneImage: 'https://picsum.photos/2400/1000',
  bannerOneLink: 'https://www.google.com',
  bannerTwo: true,
  bannerTwoName: 'Banner Two',
  bannerTwoImage: 'https://picsum.photos/600/250',
  bannerTwoLink: 'https://bluelightcard.co.uk',
  bannerThree: false,
  bannerThreeName: '',
  bannerThreeImage: '',
  bannerThreeLink: '',
};

export default componentMeta;
