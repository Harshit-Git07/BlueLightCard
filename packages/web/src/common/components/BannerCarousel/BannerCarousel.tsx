import { CampaignCard } from '@bluelightcard/shared-ui';
import React from 'react';
import SwiperCarousel from '../SwiperCarousel/SwiperCarousel';
import { BannerCarouselPropers } from './types';

const BannerCarousel = ({ banners }: BannerCarouselPropers) => {
  return (
    <SwiperCarousel
      navigation
      elementsPerPageDesktop={1}
      elementsPerPageLaptop={1}
      elementsPerPageMobile={1}
      elementsPerPageTablet={1}
    >
      {banners.map((banner, index) => (
        <CampaignCard
          {...banner}
          key={`banner-carousel-image-${index}-${banner.name}`}
          className="h-[440px]"
        />
      ))}
    </SwiperCarousel>
  );
};

export default BannerCarousel;
