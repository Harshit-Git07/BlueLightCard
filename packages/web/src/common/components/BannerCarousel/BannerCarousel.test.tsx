import { render } from '@testing-library/react';
import BannerCarousel from './BannerCarousel';
import { CampaignCardProps } from './types';

const mockBanners: CampaignCardProps[] = [
  {
    image: 'https://picsum.photos/seed/picsum/200/300',
    linkUrl: 'https://google.com',
    name: 'Mock Banner One',
  },
  {
    image: 'https://picsum.photos/seed/picsum/200/300',
    linkUrl: 'https://google.com',
    name: 'Mock Banner Two',
  },
];

describe('BannerCarousel.tsx', () => {
  it('renders', () => {
    const { container } = render(<BannerCarousel banners={mockBanners} />);
    expect(container).toMatchSnapshot();
  });
});
