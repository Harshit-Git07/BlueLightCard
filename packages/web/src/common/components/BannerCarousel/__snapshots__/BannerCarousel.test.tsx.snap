// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerCarousel.tsx renders 1`] = `
<div>
  <div
    class="swiper swiper-initialized swiper-horizontal"
  >
    <div
      class="swiper-wrapper"
      style="cursor: grab; transition-duration: 0ms; transition-delay: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
    >
      <div
        class="swiper-slide"
        data-swiper-slide-index="0"
      >
        <div
          class="relative w-full h-[440px]"
        >
          <a
            class="dark:text-palette-secondary"
            data-testid="anchor-link"
            href="https://google.com"
          >
            <img
              alt="Mock Banner One banner"
              class="w-full h-auto object-contain"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="https://picsum.photos/seed/picsum/200/300?width=3840&quality=75"
              srcset="https://picsum.photos/seed/picsum/200/300?width=640&quality=75 640w, https://picsum.photos/seed/picsum/200/300?width=750&quality=75 750w, https://picsum.photos/seed/picsum/200/300?width=828&quality=75 828w, https://picsum.photos/seed/picsum/200/300?width=1080&quality=75 1080w, https://picsum.photos/seed/picsum/200/300?width=1200&quality=75 1200w, https://picsum.photos/seed/picsum/200/300?width=1920&quality=75 1920w, https://picsum.photos/seed/picsum/200/300?width=2048&quality=75 2048w, https://picsum.photos/seed/picsum/200/300?width=3840&quality=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
            />
          </a>
        </div>
      </div>
      <div
        class="swiper-slide"
        data-swiper-slide-index="1"
      >
        <div
          class="relative w-full h-[440px]"
        >
          <a
            class="dark:text-palette-secondary"
            data-testid="anchor-link"
            href="https://google.com"
          >
            <img
              alt="Mock Banner Two banner"
              class="w-full h-auto object-contain"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="https://picsum.photos/seed/picsum/200/300?width=3840&quality=75"
              srcset="https://picsum.photos/seed/picsum/200/300?width=640&quality=75 640w, https://picsum.photos/seed/picsum/200/300?width=750&quality=75 750w, https://picsum.photos/seed/picsum/200/300?width=828&quality=75 828w, https://picsum.photos/seed/picsum/200/300?width=1080&quality=75 1080w, https://picsum.photos/seed/picsum/200/300?width=1200&quality=75 1200w, https://picsum.photos/seed/picsum/200/300?width=1920&quality=75 1920w, https://picsum.photos/seed/picsum/200/300?width=2048&quality=75 2048w, https://picsum.photos/seed/picsum/200/300?width=3840&quality=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
            />
          </a>
        </div>
      </div>
    </div>
    <div
      class="swiper-button-prev"
    />
    <div
      class="swiper-button-next"
    />
  </div>
</div>
`;
